src/prefect/cli/flow_run.py

```diff
@@ -156,9 +168,13 @@ async def logs(id: UUID):
             exit_with_error(f"Flow run {str(id)!r} not found!")
 
         while more_logs:
+            num_logs_to_return_from_page = (
+                page_size if remaining_logs is None else min(page_size, remaining_logs)
```

Comment:
I think `page_size if remaining_logs is None` might be ambiguous to me since my understanding is that this phrase means the user hasn't specified a head arg, but it doesn't mean there are no remaining logs. I think renaming `remaining_logs` to something like `user_specified_num_logs` might be clearer so that the result is more intuitive, such as `page_size if user_specified_num_logs is None`

Reply:
I agree